src/openforms/registrations/contrib/objects_api/handlers/v2.py

```diff
@@ -0,0 +1,120 @@
+"""
+Implementation details for the v2 registration handler.
+"""
+
+from collections.abc import Sequence
+from dataclasses import dataclass
+from datetime import date, datetime
+
+from glom import Assign, Path, glom
+
+from openforms.api.utils import underscore_to_camel
+from openforms.formio.typing import Component
+from openforms.typing import JSONObject, JSONValue
+
+from ..typing import ObjecttypeVariableMapping
+
+
+@dataclass
+class AssignmentSpec:
+    """
+    Describes where certain values need to be assigned in a target object.
+    """
+
+    destination: Path
+    """
+    The target path where ``value`` shall be assigned.
+    """
+    value: JSONValue
+    """
+    The value to assign at ``destination``.
+    """
+
+
+@dataclass
+class OutputSpec:
+    assignments: Sequence[AssignmentSpec]
+
+    def create_output_data(self) -> JSONObject:
+        """
+        Assigns the specified assignment configurations to the output data.
+        """
+        target: JSONObject = {}
+        spec = tuple(
+            [
+                Assign(path=assignment.destination, val=assignment.value, missing=dict)
+                for assignment in self.assignments
+            ]
+        )
+        glom(target, spec)
+        return target
+
+
+def process_mapped_variable(
+    mapping: ObjecttypeVariableMapping,
+    value: (
+        JSONValue | date | datetime
+    ),  # can't narrow it down yet, as the type depends on the component type
+    component: Component | None = None,
+) -> AssignmentSpec | Sequence[AssignmentSpec]:
+    target_path = Path(*bits) if (bits := mapping.get("target_path")) else None
+
+    # normalize non-primitive date/datetime values so that they're ready for JSON
+    # serialization in the proper format
+    if isinstance(value, (date, datetime)):
+        value = value.isoformat()
+
+    # transform the value within the context of the component
+    # TODO: convert this in a proper registry in due time so we can use better type
+    # annotations
+    match component:
+        case {"type": "addressNL"}:
+            assert isinstance(value, dict)
+            value = value.copy()
```

Comment:
Why don't we use a new variable here?I thought this is the way when we copy one.

Reply:
I'm only doing it here to avoid mutating the original object, and the type stays the same. I think either way is fine?